packages/babel-plugin/src/properties/resolve.js

```diff
@@ -42,8 +42,27 @@ export default function resolveProperty(
   }
 
   return ({ callPath, funcPath }) => {
+    const { isBrowser } = process
+
+    if (moduleFederation) {
+      if ('isBrowser' in process) {
+        if (isBrowser === true) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            'The "moduleFederation: true" option will disable code splitting on the client-side',
+          )
+        }
+      } else {
+        // eslint-disable-next-line no-console
+        console.warn(
+          'It\'s recommended to use "isBrowser" global variable to detect the environment\n' +
+            'Try to add "webpack.DefinePlugin({ \'process.isBrowser\': true })" to your webpack config',
+        )
```

Comment:
Is there something internal to a "client side" Webpack `target` that can be read automatically? For example, as a developer if I set `target: "web"` (the default) or another client-targetted value, then this plugin should probably just error out here (or have a "force" option to override).

Or does Webpack not provide a standard way to expose "is browser" and every webpack tool has to invent their own? (like [here](https://github.com/visgl/luma.gl/blob/v6.0.0/src/utils/is-browser.js))

Reply:
One can use `package.json:browser` field to detect browser target - https://github.com/theKashey/detect-node/blob/master/package.json#L6-L10
But this is a run-time operation.

Reply:
These solutions only work during runtime, @theKashey do you know if webpack provides some context to babel during the compilation?

Reply:
According to [babel-loader source](https://github.com/babel/babel-loader/blob/main/src/index.js#L54) - no, it does not.

Reply:
Hm, this begs the question then: can this be an argument to `LoadablePlugin` webpack plugin instead, and then read somehow by the Babel plugin? Is that possible? 

That would alleviate most concerns of misuse, as it could be documented to just use this option for your client-side Webpack build, which is something already natural to loadable components. 

With a Babel plugin option, users have to implement some logic to modify their Babel options dynamically based on whether it's being used by the Webpack server build or client build, which while not weird is an additional configuration workflow added to use loadable, and they also have to implement some `DefinePlugin` thing, which again is another workflow change.

No big deal, just wondering if that could simplify things

Reply:
🤔 BINGO!
`LoadablePlugin` can automatically call `DefinePlugin` with the variables required. As webpack plugin is expected to be used only for client side it can secure dead code elimination (in production only) and other checks to guide proper configuration.

Reply:
Awesome @theKashey! I will work on that here! :rocket:

Reply:
@theKashey added DefinePlugin to LoadablePlugin and updated the warning messages